src/XRP/model/xrp-payment.ts

```diff
@@ -67,21 +79,31 @@ export default class XRPPayment {
 
   /**
    *
+   * @param xrplNetwork The XRPL Network from which this object was retrieved.
    * @param amount The amount of currency to deliver.
-   * @param destination The unique address of the account receiving the payment.
-   * @param destinationTag (Optional) Arbitrary tag that identifies a hosted recipient to pay, or the reason for the payment.
+   * @deprecated @param destination The unique address of the account receiving the payment.
+   * @deprecated @param destinationTag (Optional) Arbitrary tag that identifies a hosted recipient to pay, or the reason for the payment.
    * @param deliverMin (Optional) Minimum amount of destination currency this transaction should deliver.
    * @param invoiceID (Optional) Arbitrary 256-bit hash representing a specific reason or identifier for this payment.
    * @param paths Array of payment paths to be used for this transaction.  Must be omitted for XRP-to-XRP transactions.
    * @param sendMax (Optional) Highest amount of source currency this transaction is allowed to cost.
    */
   private constructor(
+    readonly xrplNetwork: XRPLNetwork,
     readonly amount?: XRPCurrencyAmount,
     readonly destination?: string,
     readonly destinationTag?: number,
     readonly deliverMin?: XRPCurrencyAmount,
     readonly invoiceID?: Uint8Array,
     readonly paths?: Array<XRPPath | undefined>,
     readonly sendMax?: XRPCurrencyAmount,
-  ) {}
+  ) {
+    this.destinationXAddress = destination
```

Comment:
It's kinda ugly that we're going to make a public parameter be an `string | undefined` rather than a `string`, especially since this should *never* fail (destinationAddress and destinationTag should always be able to be encoded). 

I don't think we should ship this to every client of the class by having a property that is `string | undefined`. There's two options here:
1) Have this initializer throw an `unknown` exception if this ever fails. Catch exceptions in the `from` method and return `undefined` from there. 
2) We could revert to what you had before - which is maybe a bit cleaner? 

Open to suggestions here and I'm sorry for the churn.

Reply:
went with 2